src/Symfony/Bundle/FrameworkBundle/Resources/config/feature_flag.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Symfony\Component\FeatureFlag\FeatureChecker;
+use Symfony\Component\FeatureFlag\FeatureCheckerInterface;
+use Symfony\Component\FeatureFlag\Provider\ChainProvider;
+use Symfony\Component\FeatureFlag\Provider\InMemoryProvider;
+use Symfony\Component\FeatureFlag\Provider\ProviderInterface;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+
+        ->set('feature_flag.provider.in_memory', InMemoryProvider::class)
+            ->args([
+                '$features' => abstract_arg('Defined in FeatureFlagPass.'),
+            ])
+            ->tag('feature_flag.provider')
+
+        ->set('feature_flag.provider', ChainProvider::class)
+            ->args([
```

Comment:
why not `->arg('$providers', tagged_iterator('feature_flag.provider'))` ?

Reply:
By limiting to a single provider, it removes the loop from the `FeatureChecker`. Not a huge gain but using a `ChainProvider` seems an easier way to customize its behavior if any. For example, if you want to "break" this loop under certain conditions.